Content.Server/Store/Systems/StoreSystem.Refund.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Content.Server.Store.Components;
+
+namespace Content.Server.Store.Systems;
+
+public sealed partial class StoreSystem
+{
+    private void InitializeRefund()
+    {
+        SubscribeLocalEvent<StoreRefundComponent, ComponentShutdown>(OnRefundShutdown);
+    }
+
+    private void OnRefundShutdown(EntityUid uid, StoreRefundComponent component, ComponentShutdown args)
+    {
+        if (MetaData(uid).EntityLifeStage == EntityLifeStage.Terminating)
+        {
+            var ev = new RefundEntityDeletedEvent(uid);
+            RaiseLocalEvent(component.StoreEntity, ev);
+        }
+    }
```

Comment:
EntityTerminatingEvent is raised directed so use that instead, you will also need to check if the storeentity itself is deleting in case this is called as part of the round restarting or otherwise.